sdk/ai/azure-ai-projects/samples/agents/sample_agents_openapi_morningstar.py

```diff
@@ -0,0 +1,191 @@
+# pylint: disable=line-too-long,useless-suppression
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+"""
+DESCRIPTION:
+    This sample demonstrates how to use agent operations with the
+    Morningstar OpenAPI tool, (https://www.morningstar.com/) from the Azure Agents service using a synchronous client.
+    To learn more about OpenAPI specs, visit https://learn.microsoft.com/openapi
+
+USAGE:
+    python sample_agents_openapi_morningstar.py
+
+    Before running the sample:
+
+    pip install azure-ai-projects azure-identity jsonref
+
+    Set these environment variables with your own values:
+    1) PROJECT_CONNECTION_STRING - The project connection string, as found in the overview page of your
+       Azure AI Foundry project.
+    2) MODEL_DEPLOYMENT_NAME - The deployment name of the AI model, as found under the "Name" column in
+       the "Models + endpoints" tab in your Azure AI Foundry project.
+    3) PROJECT_CONNECTION_STRING - The project connection string of your AI Foundry project.
+
+    You'll also need a JWT token to authorize with the Morningstar service.
+    You can fetch this JWT token by using the below code:
+
+    import requests
+    from requests.auth import HTTPBasicAuth
+
+    url = "https://www.us-api.morningstar.com/token/oauth"
+    auth = HTTPBasicAuth("john.doe@example.com", "SecureP@ssw0rd")
+    response = requests.post(url, auth=auth)
+    print(response.text)
+
+    Note: This token expires, please remember to refresh your credentials.
+"""
+
+import os
+import jsonref
+from azure.ai.projects import AIProjectClient
+from azure.ai.projects.models import (
+    OpenApiConnectionAuthDetails,
+    OpenApiConnectionSecurityScheme,
+    OpenApiTool,
+)
+from azure.identity import DefaultAzureCredential
+
+project_client = AIProjectClient.from_connection_string(
+    credential=DefaultAzureCredential(),
+    conn_str=os.environ["PROJECT_CONNECTION_STRING"],
+)
+# [START create_agent_with_openapi]
```

Comment:
Delete this comment (it was used in the original sample to embed its code in the package README.md)

Reply:
Done